06_puppy/sirigithub/stores_test.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type storesSuite struct {
+	suite.Suite
+	store     Storer
+	newStorer func() Storer
+	puppy1    Puppy
+}
+
+func (suite *storesSuite) SetupTest() {
+	suite.store = suite.newStorer()
+	suite.puppy1 = Puppy{ID: 1, Breed: "cavalier", Color: "White", Value: "$1700"}
+	er := suite.store.CreatePuppy(&suite.puppy1)
+	if er != nil {
+		suite.FailNow("Test setup failed to create puppy", er)
+	}
+
+}
+func (suite *storesSuite) TestCreatePuppySuccesful() {
+	r := require.New(suite.T())
+	newPuppy := Puppy{ID: 2, Breed: "Dobberman", Color: "Black", Value: "$800"}
+	er := suite.store.CreatePuppy(&newPuppy)
+	r.NoError(er, "Create should not produce an error")
+}
+func (suite *storesSuite) TestReadPuppySuccesful() {
+	//given
+	assert := assert.New(suite.T())
+	r := require.New(suite.T())
+	//when
+	puppy, err := suite.store.ReadPuppy(suite.puppy1.ID)
+	//then
+	r.NoError(err, "Read should not produce an error")
+	assert.Equal(puppy, &suite.puppy1)
+}
+func (suite *storesSuite) TestCreatePuppyError() {
+	//given
+	assert := assert.New(suite.T())
+	//when
+	createEr := suite.store.CreatePuppy(&suite.puppy1)
+	//then
+	assert.Error(createEr, "Cannot create existing puppy")
+}
+func (suite *storesSuite) TestReadPuppyDoesNotExist() {
```

Comment:
blank line between functions.

Reply:
fixed.

Reply:
fixed

Reply:
Really?
I don't see any blank line ;)

Reply:
Please fix below too.

Reply:
As Julia asked, a blank line between functions makes this consistent with the other files and easier to scan-read.

Reply:
included blank lines between functions.